soql-standalone-parser/src/main/scala/com/socrata/soql/ast/Select.scala

```diff
@@ -357,6 +390,23 @@ case class Selection(allSystemExcept: Option[StarSelection], allUserExcept: Seq[
 
   def replaceHoles(f: Hole => Expression) =
     copy(expressions = expressions.map(_.replaceHoles(f)))
+
+  def doc = docs.punctuate(d",").hsep
+
+  def docs: Seq[Doc[Nothing]] = {
+    val docs = Seq.newBuilder[Doc[Nothing]]
+    docs ++= allSystemExcept.map(_.doc(":*"))
+    docs ++= allUserExcept.map(_.doc("*"))
+    docs ++= expressions.map { case SelectedExpression(expr, name) =>
+      name match {
+        case Some((n, _)) =>
+          d"${expr.doc} AS `${n.toString}`"
```

Comment:
How come this string gets formatted normally, not through docs?

Reply:
The `d` interpolator can do that for you.  I was experimenting with that and then decided I preferred `d` as _just_ a string literal syntax, but it actually is a full interpolator.  This is precisely equivalent to ```expr.doc ++ d" AS `" ++ Doc(n.toString) ++ d"`"```.  I'll probably go back and change it to the other style.